pinot-query-planner/src/main/java/org/apache/pinot/query/planner/logical/PlanFragmenter.java

```diff
@@ -89,12 +89,26 @@ private PlanNode process(PlanNode node, Context context) {
 
   @Override
   public void onSubstitution(int receiver, int oldSender, int newSender) {
+    // Change the sender of the receiver to the new sender
     IntList senders = _childPlanFragmentIdsMap.get(receiver);
     senders.rem(oldSender);
     if (!senders.contains(newSender)) {
       senders.add(newSender);
     }
+
+    // Remove the old sender and its children from the plan fragment map
```

Comment:
This can be removed now since it'll be done in the first iteration of the loop right?

Reply:
but it still applies right?